src/morphology/english/determineStem.js

```diff
@@ -13,17 +13,26 @@ import { flatten } from "lodash-es";
  *
  * @returns {string} The shortest of the alphabetically ordered strings from the input array.
```

Comment:
It can return `undefined` on an empty array too.
Maybe you'd want to return an empty string then like in `determineIrregularStem`.
I'm not sure which is better, it depends if you need to be able to check if that happened or not. And if an empty string works fine for code using that as a valid result.
Either way, it is probably nice to be consistent here.

Reply:
After serious consideration, I decided I want to return `undefined` if any of the sub-functions fails to return a string base. 
I think it's much more valid than to return an empty string, because we definitely don't want empty strings to invade databases. `getInfinitive` and `getBase` functions used in `determineRegularStem` both return the input word as default, which makes them pretty safe and the chances to get an empty array as input to `findShortestAndAlphabeticallyFirst` and to return `underfined` by `defineStem` indefinitely close to 0. And if it ever happens I want to get an error and not to stuff empty strings in the databases.